### PR TITLE
Add Clipboard.Clear(); at the end of the tests

### LINKED ISSUE #12002: Clipboard Tests still failing
### Test

System.Windows.Forms.Tests.TextBoxBaseTests+ClipboardTests.TextBoxBase_ClearUndo_CanUndo_Success

It's very random as to which tests fail and how many, I have seem between 0 and 6 on GitHub and locally. Even if you just manually run the tests in just 1 class.

### Issue description

Failing Clipboard tests are causing build to fail.

### PR DESCRIPTION
**Pull request by LeafShi1**
Related #12002

## Proposed changes

- Add `Clipboard.Clear();` at the begin of all tests
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12032)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs [hunk 1]
```diff
@@ -127,86 +122,82 @@ public void TextBoxBase_Cut_PasteNotEmptyWithHandle_Success()
...
+            control.IsHandleCreated.Should().BeTrue();
+            invalidatedCallCount.Should().Be(0);
+            styleChangedCallCount.Should().Be(0);
+            createdCallCount.Should().Be(0);
         }
 
         [WinFormsFact]
         public void TextBoxBase_Paste_InvokeEmpty_Success()
         {
+            Clipboard.Clear();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:148]: I hadn't seen any improvements from adding Clear to the tests in the past. This change is increasing chances of Clipboard methods invoked by other test or application to fails, and in the case of winforms managed APIs, SetDataObject method will replace the empty object or any other object on the clipboard, content does not matter as much as the fact of access to a single OS-wide resource does. 

But more interestingly, are you sure that this Paste operation touches the clipboard? I don't see any OLE APIs invoked https://github.com/dotnet/winforms/blob/8cdb3e05cad0f6c43e336b42f2cdefd337c8ed54/src/System.Windows.Forms/src/System/Windows/Forms/Controls/TextBox/TextBoxBase.cs#L977 nor do I see any OLE APIs when the comctl32 control processes EM_REPLACESEL message. It is simply about setting control text to a new value taking into account max size.

What kind of failures had you seen that prompted you to make this change?

**Review by Tanya-Solyanik — COMMENTED** [review]
Please pull out the move to fluent assertions into a new PR and we'll merge it. Otherwise I don't see clipboard operations in this file, please share the test failures that pointed to clipboard in these tests. I think this file should not be in the sequential collection. Additionally I don't think that `Clear` improves test reliability, rather it adds pressure on the singleton resource (Clipboard)

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs (hunk 1 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:148]: @Tanya-Solyanik this fix was related to issue #12002 but I do not see how Clipboard.Clear would fix that issue.

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs [hunk 2]
```diff
@@ -127,86 +122,82 @@ public void TextBoxBase_Cut_PasteNotEmptyWithHandle_Success()
...
             using SubTextBox control = new();
             control.Paste();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:150]: I see, this is a different override of Paste, the one that takes no arguments :). Sure, Clipboard.Clear is required in this case.

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs [hunk 3]
```diff
@@ -8,59 +8,60 @@ public partial class TextBoxBaseTests
     [Collection("Sequential")]
     public class ClipboardTests
     {
-        [WinFormsFact]
-        public void TextBoxBase_ClearUndo_CanUndo_Success()
+        private static SubTextBox CreateSubTextBox()
         {
-            using SubTextBox control = new()
+            Clipboard.Clear();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:13]: Please add `Clipboard.Clear` only to tests that don't do `Copy`

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs (hunk 1 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:148]: Never mind, I looked up a wrong method.

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs [hunk 4]
```diff
@@ -127,45 +131,48 @@ public void TextBoxBase_Cut_PasteNotEmptyWithHandle_Success()
...
         {
+            Clipboard.Clear();
             using SubTextBox control = new()
             {
                 Text = "abc",
                 SelectionStart = 1,
                 SelectionLength = 2
             };
             control.Paste();
-            Assert.Equal("abc", control.Text);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:167]: How was this working before this fix?

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs (hunk 4 above)
Comment (LeafShi1) [reply src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:167]: > How was this working before this fix?

Before adding Clipboard.Clear(), the clipboard retains the `bc` cut by the previous case, so this case can run through, but once the running order of cases is adjusted, this case will definitely fail, so according to the previous code, these cases are not independent.

Take the case `TextBoxBase_Paste_InvokeEmpty_Success ` as an example to better illustrate the problem
Before changes:
image

After change:
image

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.ClipboardTests.cs:167]: >Before adding Clipboard.Clear(), the clipboard retains the bc cut by the previous case,

Great catch!

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Pull request merged**